proposals/lotus-dynamic-retrieval-pricing.md

```diff
@@ -0,0 +1,148 @@
+# Dynamic retrieval pricing in Lotus
+
+Authors: @raulk
+
+Initial PR: https://github.com/protocol/web3-dev-team/pull/97.
+
+## Purpose &amp; impact 
+#### Background &amp; intent
+_Describe the desired state of the world after this project? Why does that matter?_
+
+A retrieval client asks the miner for a quote before proceeding to retrieve a
+piece or a part therein. The miner quotes:
+
+- unseal price (the price the client must pay if the miner needs to unseal data)
+- minimum price per byte (the price the client must pay for each byte of data)
+- maximum payment interval (the number of bytes that the provider will send
+  before pausing the transfer to wait for payment)
+- maximum payment interval increase (the increase in the payment interval each
+  time payment is received)
+
+The quoted values become part of the deal proposal. Unlike with storage deals,
+retrieval deal proposals don’t go on chain.	
+
+On the provider side, these values are filled from the “stored ask”.
+The “stored ask” can be changed through JSON-RPC, but it’s statically set.
+Lotus doesn’t have the ability to change the quote based on the piece that’s
+being requested.
+
+This means that fast retrieval and non-fast retrieval deals get the same
+treatment with regards to pricing. In other words, anything that’s set as the
+"stored ask" will be a blanket policy for all retrievals, fast or ordinary,
+verified or unverified.
+
+Impact of the blanket policy to unsealing price:
+- If the unsealing price is set to zero, the miner commits to serving all deals
+  without requiring an unseal payment from the client, even those that require
+  unsealing. Unsealing is a heavy and expensive operation, one that's irrational
+  for miners to offer for free today.
+- If the unsealing price is set to non-zero, fast retrievals will require the
+  creation of a payment channel (even if price per byte is zero), which
+  introduces significant latency and requires chain interaction.
+
+Impact of the blanket policy to price per byte:
+- Miners wanting to provide free retrieval for verified deals would set this
+  value to zero. This has the side-effect of skipping payment channel creation 
+  (if unseal price is also zero).
+- However, that same policy would apply to unverified deals, which the miner is
+  likely not keen to offer for free.
```

Comment:
Hm... this seems incongruent with what the goldenpath around retrieval seems to be?  That is: mining is so profitable that we do expect  in the short to medium term (multiple year) that miners will be offering free retrieval provided Fil+ is in place. Also dealbot efforts are centered around making unsealing untenable, by effectively requiring everything to be stored twice ( again - miners will go with it because it is cheaper to store anyway ).

Did thinking really shift that much since mid-march?

Reply:
The goal hasn't changed. This line says that, currently, setting the retrieval price to byte to zero affects all storage deals, both verified AND unverified. That's one of the things we need to make dynamic.

Reply:
I should have done a more precise highlight. Let me clarify:

- On unsealing: my understanding is that there is broad agreement ( including from CE ) that the act of unsealing is uneconomical given the current proofs system, and thus the option ought to be deprioritized if not deprecated within the "happy path". That is - miner always stores both sectors ( which is currently switched to be the default ) and if they do elect to delete some of the unsealed replicas, it is on them to pay the price for recovery. Additionally the dealbots ranking of TTFB will further make not providing fast retrieval untenable. Thus it seems odd at present expending effort and API-surface on making unsealing more visible.

- My other point was on how free and non-free retrieval are listed as equal options in the highlighted text, whereas it would be prudent to "nudge" both miner and client API users / implementors to gravitate towards "free by default, and not doing so is somewhat of a chore". In the spirit of "defaults matter" etc.